AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -204,6 +217,11 @@ - (ASEnvironmentTraitCollection)environmentTraitCollection
   return _environmentState.traitCollection;
 }
 
+- (void)setEnvironmentTraitCollection:(ASEnvironmentTraitCollection)traitCollection
+{
+  _environmentState.traitCollection = traitCollection;
```

Comment:
@rcancro that all makes sense - I definitely agree that ASTraitCollection should be the public-facing API, and the struct is an implementation technique chosen for a variety of advantages.

However, I am confused at the existence of a dedicated ivar for _traitCollection that appears to be able to get out of sync with the struct.  My assumption with the strategy to create an object-based trait collection was that it would be created on demand when accessed, and if it were cached at all, would at least be nil'd out when the underlying struct changes (then regenerated upon the next access of the object).

As it is written now (not only this method but also looking at the one above), wouldn't it be possible to access the two types of trait collections, compare the values, and find some are not equal?  This seems like it should not be possible.